### PR TITLE
Add integration tests

### PR DESCRIPTION
**Pull request by r2d4**

### CODE REVIEW

**Comment by r2d4** [comment]
I'm working on exposing the logs publicly, but this should be ready now. I'll follow up with some docs to run this locally with a GKE cluster or minikube.

File: deploy/skaffold/Dockerfile [hunk 1]
```diff
@@ -12,19 +12,24 @@
...
+        curl \
+        build-essential \
+        git \ 
+        gcc \
+        python-dev \
+        python-setuptools \
+        apt-transport-https \
+        lsb-release
+
+COPY --from=golang:1.10 /usr/local/go /usr/local/go
```
Comment (dlorenc) [inline deploy/skaffold/Dockerfile]: Go isn't in the ubuntu package manager anywhere?

File: deploy/skaffold/Dockerfile [hunk 2]
```diff
@@ -35,13 +40,28 @@ RUN curl -LO https://storage.googleapis.com/kubernetes-helm/helm-${HELM_VERSION}
     tar -zxvf helm-${HELM_VERSION}-linux-amd64.tar.gz && \
     mv linux-amd64/helm /usr/local/bin/helm
 
-FROM gcr.io/distroless/base 
-WORKDIR /root/
+ENV CLOUD_SDK_VERSION 193.0.0
+RUN easy_install -U pip && \
+    pip install -U crcmod && \
```
Comment (dlorenc) [inline deploy/skaffold/Dockerfile]: Do we use gsutil anywhere? We only need crcmod to make some gsutil operations faster.

File: deploy/skaffold/Dockerfile [hunk 3]
```diff
@@ -35,13 +40,28 @@ RUN curl -LO https://storage.googleapis.com/kubernetes-helm/helm-${HELM_VERSION}
...
+    export CLOUD_SDK_REPO="cloud-sdk-$(lsb_release -c -s)" && \
+    echo "deb https://packages.cloud.google.com/apt $CLOUD_SDK_REPO main" > /etc/apt/sources.list.d/google-cloud-sdk.list && \
```
Comment (dlorenc) [inline deploy/skaffold/Dockerfile]: I'm not a big fan of using the deb installer for the SDK, it makes extra components weird. Can you use the tarball instead? https://cloud.google.com/sdk/downloads#versioned

File: integration/run_test.go [hunk 4]
```diff
@@ -19,31 +19,58 @@ limitations under the License.
...
 
+var gkeZone = flag.String("gke-zone", "us-central1-a", "gke zone")
+var gkeClusterName = flag.String("gke-cluster-name", "integration-tests", "name of the integration test cluster")
+var gcpProject = flag.String("gcp-project", "k8s-skaffold", "the gcp project where the integration test cluster lives")
+var remote = flag.Bool("remote", false, "if true, run tests on a remote GKE cluster")
+
+func TestMain(m *testing.M) {
+	flag.Parse()
+	if *remote {
+		cmd := exec.Command("gcloud", "container", "clusters", "get-credentials", *gkeClusterName, "--zone", *gkeZone, "--project", *gcpProject)
```
Comment (dlorenc) [inline integration/run_test.go]: Should we use our exec.Command helper instead?

File: deploy/skaffold/Dockerfile (hunk 1 above)
Comment (r2d4) [reply deploy/skaffold/Dockerfile]: We can't ensure that we use the same version that way

File: deploy/skaffold/Dockerfile (hunk 2 above)
Comment (r2d4) [reply deploy/skaffold/Dockerfile]: No, I can remove this. Wasn't sure if we were going to need upload anything once we were inside running the tests

File: integration/run_test.go (hunk 4 above)
Comment (r2d4) [reply integration/run_test.go]: The helper is used on the next line below, we don't do anything special for creating the cmd, just running

File: deploy/skaffold/Dockerfile (hunk 1 above)
Comment (dlorenc) [reply deploy/skaffold/Dockerfile]: Ah, makes sense.

File: integration/run_test.go (hunk 4 above)
Comment (dlorenc) [reply integration/run_test.go]: D'oh, I was thinking of the special one we have in minikube for integration tests. Nevermind.

File: Makefile [hunk 5]
```diff
@@ -82,3 +87,16 @@ release: cross
...
+
+.PHONY: integration-in-docker
+integration-in-docker:
+	docker build \
+		-f deploy/skaffold/Dockerfile \
+		-t gcr.io/$(GCP_PROJECT)/skaffold-integration .
+	docker run \
+		-v /var/run/docker.sock:/var/run/docker.sock \
+		-v $(HOME)/.config/gcloud:/root/.config/gcloud \
+		-v $(PWD):/go/src/github.com/GoogleCloudPlatform/skaffold \
```
Comment (dlorenc) [inline Makefile]: nit: reuse the variables for GoogleCloudPlatform/skaffold?

File: deploy/skaffold/Dockerfile (hunk 3 above)
Comment (r2d4) [reply deploy/skaffold/Dockerfile]: Yeah, I'll do that

File: deploy/skaffold/Dockerfile (hunk 2 above)
Comment (r2d4) [reply deploy/skaffold/Dockerfile]: fixed

File: Makefile (hunk 5 above)
Comment (r2d4) [reply Makefile]: fixed

**Comment by r2d4** [comment]
This passed, re-running with an internal config change for public logs.

**Comment by r2d4** [comment]
@dlorenc I think this is ready to merge.

A note to users, we're working on getting approval to make the logs public, please bear with us. In the meantime, I'll make sure that there are clear directions on how to run the tests locally.

**Pull request merged**